### DISCUSSION TITLE
App Check with App Distribution

### DISCUSSION THREAD

**Discussion by riccardocescon** [firebase/flutterfire#10908] [Q&A]
I have enabled AppCheck on the app i am working on.
If i register the Debug Token everything works perfectly.
If i load the release with `flutter run --release` command, i am not able to use the app due to App Check because it rejects all requests marking them as 'not valid' on the App Check page on Firebase.
If i publish the app on the play console on the Internal Tester and then download the app from there, then i am able to use it.
Isn't the `flutter run --release` command compiling and singing the apk before loading to my connected physical device?
And then, how should it connect with App Distribution?
Since it's not the play console store, i just load the apk or app bundle generated with flutter` build apk --release` and it won't work cause of App Check, so is it possible to use both App Check and App Distribution for internal testing?

**Comment by danielgomezrico** [comment]
Im wondering that if you deploy with FIrebase App Distribution and enable AAB deployments, maybe those uses the key from the play store to do the releases?

https://firebase.google.com/docs/app-distribution/android/distribute-console?hl=en&authuser=0&_gl=1*1uar9ll*_ga*ODA5NDE2NjczLjE3MzY3MTE0NzA.*_ga_CW55HF8NVT*MTc0MDY2OTUwMy4xMTcuMS4xNzQwNjcwMTU5LjU4LjAuMA..&apptype=aab

Im having the same trouble but I can not test it right now, I dont have enough permissions on the play store.
